app/repositories/mails.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import textwrap
+from typing import cast
+from typing import TypedDict
+
+import app.state.services
+
+# +--------------+------------------------+------+-----+---------+-------+
+# | Field        | Type                   | Null | Key | Default | Extra |
+# +--------------+------------------------+------+-----+---------+-------+
+# | id           | int                    | NO   | PRI | NULL    |       |
+# | from_id      | int                    | NO   |     | NULL    |       |
+# | to_id        | int                    | NO   |     | NULL    |       |
+# | msg          | varchar(2048)          | NO   |     | NULL    |       |
+# | time         | int                    | YES  |     | NULL    |       |
+# | read         | tinyint(1)             | NO   |     | NULL    |       |
+# +--------------+------------------------+------+-----+---------+-------+
+
+
+class Mail(TypedDict):
+    id: int
+    from_id: int
+    to_id: int
+    msg: str
+    time: int
+    read: bool
+
+
+READ_PARAMS = textwrap.dedent(
+    """\
+        id, from_id, to_id, msg, time, `read`
+    """,
+)
+
+
+async def create(from_id: int, to_id: int, msg: str) -> Mail:
+    """Create a new mail entry in the database."""
+    query = f"""\
+        INSERT INTO mail (from_id, to_id, msg, time)
+             VALUES (:from_id, :to_id, :msg, UNIX_TIMESTAMP())
+    """
+    params = {"from_id": from_id, "to_id": to_id, "msg": msg}
+    rec_id = await app.state.services.database.execute(query, params)
+
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM mail
+         WHERE id = :id
+    """
+    params = {
+        "id": rec_id,
+    }
+    mail = await app.state.services.database.fetch_one(query, params)
+
+    assert mail is not None
+    return cast(Mail, dict(mail._mapping))
+
+
+async def fetch_count(
+    from_id: int | None = None,
+    to_id: int | None = None,
+    time: str | None = None,
+    read: bool | None = None,
+) -> int:
```

Comment:
the interface of fetch_count should always match the interface of the fetch_many provided. in this case fetch_many is fetch_all, but same rule should apply.

however, i don't think we need fetch_count here for this use case, so i think we can just remove this